oidc-client/src/Auth/OIDCUserProvider.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+
+namespace GCS\OIDCClient\Auth;
+
+
+use Illuminate\Contracts\Auth\Authenticatable;
+use Illuminate\Contracts\Auth\UserProvider;
+
+
+class OIDCUserProvider implements UserProvider
+{
+    private function is_admin($user_info)
+    {
+	    if (!property_exists($user_info, "resource_access"))
+		    return false;
+	    if (!property_exists($user_info->resource_access, "pixelfed"))
+		    return false;
+	    if (!property_exists($user_info->resource_access->pixelfed, "roles"))
+		    return false;
+	    return in_array("admin", $user_info->resource_access->pixelfed->roles);
```

Comment:
These are specific to how you configure keycloak, and can't be relied on for OIDC, the Mastodon PR had a similar issue too. (If you want a different provider to test with try Ory.sh)